crates/turborepo-lib/src/run/scope/change_detector.rs

```diff
@@ -28,10 +30,10 @@ impl<'a> ScopeChangeDetector<'a> {
         turbo_root: &'a AbsoluteSystemPath,
         scm: &'a SCM,
         pkg_graph: &'a PackageGraph,
-        global_deps: Vec<String>,
         ignore_patterns: Vec<String>,
     ) -> Self {
-        let change_mapper = ChangeMapper::new(pkg_graph, global_deps, ignore_patterns);
+        let pkg_detector = DefaultPackageDetector::new(pkg_graph);
+        let change_mapper = ChangeMapper::new(pkg_graph, ignore_patterns, pkg_detector);
```

Comment:
In theory it's ok since if you're changing a root file that's not a globalDependency, that should result in a cache hit, but I think we should discuss this before doing anything